src/main/scala/edu/twitter/model/impl/GradientBoostingBuilder.scala

```diff
@@ -23,6 +26,9 @@ class GradientBoostingBuilder(sc: SparkContext) extends GenericModelBuilder {
     * @return GenericModel
     */
   def build(): GenericModel = {
+    implicit def toRecords(data: RDD[(String, LabeledPoint)]): RDD[Record] = data.map { case (msg, p) => Record(msg, p.label) }
+    implicit def toLabeledPoint(data: RDD[(String, LabeledPoint)]): RDD[LabeledPoint] = data.map(_._2)
```

Comment:
Sorry I couldn't find where we use this?

Reply:
they are being used in converting `RDD[(String, LabeledPoing)]` to `RDD[LabeledPoint]` and RDD[Record]`. the scala compiler will invoke the method when it needs the target type. you can find more about `implicits` in scala [here](https://docs.scala-lang.org/tour/implicit-conversions.html).